File: swift/04.swift
func isPalindrome(_ n: Int) -> Bool {
    return n == Int(String(String(n).reversed()))
}


var largest = 0

for i in 100..<1000 {
    for j in 100..<1000 {
        if isPalindrome(i * j) && i * j > largest {
            largest = i * j
        }
    }
}

print(largest)

File: python/04.py
def is_palindrome(n):
    return n == int(str(n)[::-1])


largest = 0

for i in range(100, 1000):
    for x in range(100, 1000):
        if is_palindrome(i * x) and i * x > largest:
            largest = i * x

print(largest)
